polygeist/test/polygeist-opt/sycl/sycl-raise-host.mlir

```diff
@@ -0,0 +1,16 @@
+// RUN: polygeist-opt --sycl-raise-host --split-input-file %s | FileCheck %s
+
+gpu.module @device_functions {
+  gpu.func @foo() kernel {
+    gpu.return
+  }
+}
+
+// CHECK-LABEL: sycl.host.kernel_name @kernel_ref -> @device_functions::@foo
+llvm.mlir.global private unnamed_addr constant @kernel_ref("foo\00") {addr_space = 0 : i32, alignment = 1 : i64, dso_local}
+
+llvm.func @f() -> !llvm.ptr {
```

Comment:
Create another function that uses the global to make sure we don't end up removing it if it is still needed.